signal_timeout_block.py

```diff
@@ -67,8 +68,10 @@ def start(self):
                         # scheduled before this persisted job is scheduled, so
                         # only do this if one has not been scheduled already.
                         self._schedule_timeout_job(job, key, interval, True)
+        self._persistence_scheduled.set()
```

Comment:
Obviously the performance hit caused by this PR isn't very much, but checking an event for every signal that comes through solely for the purpose of solving a race condition on the block startup feels a bit off to me.